commercetools/machine-learning/commercetools-ml-api/src/main/java/com/commercetools/api/ml/generated/client/ByProjectKeyImageSearchPost.java

```diff
@@ -0,0 +1,156 @@
+package com.commercetools.api.ml.generated.client;
+
+import io.vrap.rmf.base.client.utils.Utils;
+import io.vrap.rmf.base.client.utils.json.VrapJsonUtils;
+
+import java.io.InputStream;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+import java.util.concurrent.CompletableFuture;
+import io.vrap.rmf.base.client.utils.Generated;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import io.vrap.rmf.base.client.*;
+
+
+/**
+*  <p>Accepts an image file and returns similar products from product catalogue.</p>
+*/
+@Generated(
+    value = "io.vrap.rmf.codegen.rendring.CoreCodeGenerator",
+    comments = "https://github.com/vrapio/rmf-codegen"
+)
+public class ByProjectKeyImageSearchPost {
+   
+   
+   private ApiHttpHeaders headers = new ApiHttpHeaders();
+   private Map<String, String> additionalQueryParams = new HashMap<>();
+   private final ApiHttpClient apiHttpClient; 
+   private List<Integer> limit = new ArrayList<>();
+   private List<Integer> offset = new ArrayList<>();
+   private String projectKey;
+   
+   private java.io.File file;
+   
+   public ByProjectKeyImageSearchPost(final ApiHttpClient apiHttpClient, String projectKey, java.io.File file){
+      this.apiHttpClient = apiHttpClient;
+      this.projectKey = projectKey;
+      this.file = file;
+   }
+   
+   public ApiHttpRequest createHttpRequest() {
+      ApiHttpRequest httpRequest = new ApiHttpRequest();
+      List<String> params = new ArrayList<>();
+      params.add(this.limit.stream().map(s -> "limit=" + s).collect(Collectors.joining("&")));
+      params.add(this.offset.stream().map(s -> "offset=" + s).collect(Collectors.joining("&")));
+      params.add(additionalQueryParams.entrySet().stream().map(entry -> entry.getKey() + "=" + entry.getValue()).collect(Collectors.joining("&")));
+      params.removeIf(String::isEmpty);
+      String httpRequestPath = String.format("/%s/image-search", this.projectKey);
+      if(!params.isEmpty()){
+         httpRequestPath += "?" + String.join("&", params);
+      }
+      httpRequest.setRelativeUrl(httpRequestPath); 
+      httpRequest.setMethod(ApiHttpMethod.POST);
+      httpRequest.setHeaders(headers);
+      try{httpRequest.setBody(VrapJsonUtils.toJsonByteArray(file));}catch(Exception e){e.printStackTrace();}
```

Comment:
This looks interesting 😉  May be we should improve the API later, doesn't seem to be the most efficient way to  transfer an image file 😉

Reply:
I think it doesn't even work in that case, i added a follow up issue for that https://github.com/vrapio/rmf-codegen/issues/149